src/smart-components/role/add-role-new/add-permissions.js

```diff
@@ -0,0 +1,115 @@
+        operations: [ ...new Set(data.filter(permission => filterApplication(permission) && filterResource(permission)).map(({ operation }) => operation)) ]
+    };
+};
+
+const AddPermissionsTable = (props) => {
+    const dispatch = useDispatch();
+    const fetchData = () => dispatch(getPrincipalAccess());
+    const { access, isLoading } = useSelector(selector, shallowEqual);
+    const { input } = useFieldApi(props);
+    const [ permissions, setPermissions ] = useState({ filteredData: [], applications: [], resources: [], operations: []});
+    const [ filters, setFilters ] = useState({ applications: [], resources: [], operations: []});
+    const [ selectedPermissions, setSelectedPermissions ] = useState([]);
+    const [ pagination, setPagination ] = useState({ limit: 10, offset: 0 });
+
+    const createRows = (permissions) => permissions.map(
+        ({ application, resource, operation, uuid }) => ({
+            uuid: `${application}:${resource}:${operation}`,
+            cells: [
+                application,
+                resource,
+                operation
+            ],
+            selected: Boolean(selectedPermissions && selectedPermissions.find(row => row.uuid === uuid))
+        })
+
+    );
+
+    useEffect(() => {
+        fetchData();
+    }, []);
+
+    useEffect(() => {
+        setPermissions(accessWrapper(access, filters));
+    }, [ access ]);
+
+    useEffect(() => {
+        setPermissions(accessWrapper(access, filters));
+    }, [ filters ]);
+
+    useEffect(() => {
+        input.onChange(selectedPermissions);
+    }, [ selectedPermissions ]);
+
+    const setCheckedItems = (newSelection) => {
+        setSelectedPermissions(newSelection(selectedPermissions).map(({ uuid }) => ({ uuid })));
+    };
+
+    return <div>
+        <TableToolbarView
+            columns={ columns }
+            isSelectable={ true }
+            isCompact={ true }
+            borders={ false }
+            createRows={ createRows }
+            data={ permissions.filteredData.slice(pagination.offset, pagination.offset + pagination.limit) }
+            filterValue={ '' }
+            fetchData={ ({ limit, offset }) => setPagination({ limit, offset }) }
+            setFilterValue={ ({ applications, resources, operations }) => {
+                typeof applications !== 'undefined' && setFilters({ ...filters, applications });
+                typeof resources !== 'undefined' && setFilters({ ...filters, resources });
+                typeof operations !== 'undefined' && setFilters({ ...filters, operations });
+            } }
+            isLoading={ isLoading }
+            pagination={ { ...pagination, count: permissions.filteredData.length } }
+            checkedRows={ selectedPermissions }
+            setCheckedItems={ setCheckedItems }
+            titlePlural="permissions"
+            titleSingular="permission"
+            hideFilterChips
+            textFilters={ [
```

Comment:
Good for now, but we might want to reconsider changing this prop to `filters` instead of `textFilters`

Reply:
Yep, I agree. But I would do it in separate PR.